src/append_only_data.rs

```diff
@@ -431,15 +498,18 @@ macro_rules! impl_appendable_data {
                 })
             }
 
+            /// Gets the last entry.
             fn last_entry(&self) -> Option<&Entry> {
                 self.inner.data.last()
             }
 
+            /// Gets a complete list of permissions.
             fn permissions(&self, index: impl Into<Index>) -> Option<&P> {
                 let index = to_absolute_index(index.into(), self.inner.permissions.len())?;
                 self.inner.permissions.get(index)
             }
 
+            /// Returns the owner key.
```

Comment:
It's not entirely clear what `key` means in this context, I'd suggest replacing it with something like
```suggestion
            /// Returns the owner's public key and the indices at the time it was added.
```